build.gradle

```diff
@@ -163,18 +241,123 @@ task opensslTest(type: Test) {
 }
 
 task copyExtraTestResources(dependsOn: testClasses) {
+
     copy {
         from 'src/test/resources'
         into 'build/testrun/test/src/test/resources'
     }
+
+    taskNames.each { testName ->
+        copy {
+            from 'src/test/resources'
+            into "build/testrun/${testName}/src/test/resources"
+        }
+    }
+
+    copy {
+        from 'src/test/resources'
+        into 'build/testrun/citest/src/test/resources'
+    }
+}
+
+task citest(type: Test) {
```

Comment:
This task seems like it is duplicated with the section in 312, can we trim this down by declaring and reusing the test configuration or something like that?

Reply:
No, it is not. `citest` is "generic" one which covers everything not covered by specific tests (declared in `splitTestConfig` object). Loop on line 312 is generating tasks for `splitTestConfig`.

Reply:
True, what about the parameters after `filter`?  seems like there is 20+ lines of code that are repeated in 2+ places that makes the layout of these look similar and changes should be consistent across

Reply:
> True, what about the parameters after `filter`? seems like there is 20+ lines of code that are repeated in 2+ places that makes the layout of these look similar and changes should be consistent across


I have extracted the parts related to configuration into a separate function called `setCommonTestConfig` in order to avoid code duplication.

Reply:
@pchmielnik's changes has been applied on repository.